Webscraper

mugicha101 opened the pull request:
**Overview**
Uses a webscraper to handle queries to the prison state websites and stores the response in a csv

**Files added in backend/webscraper/**
- state_websites.py - holds classes that handle specific state websites
- webscraper.py - handles lists of queries concurrently and stores responses in a csv file (currently output.csv)
- csv_handler.py - handles output to csv
- test.py - sample/test driver code that shows how to utilize the web-scraper
- requirements.txt - stores python dependencies needed for the webscraper to work
- README.txt - shows how to install dependencies
- output.csv - stores query responses

**Python Dependencies (also in requirements.txt)**
- beautifulsoup4
- urllib3

mugicha101 committed:
test commit

mugicha101 committed:
basic structure implemented

mugicha101 committed:
added comments

mugicha101 committed:
minor change

mugicha101 committed:
restructuring

mugicha101 committed:
Delete backend/webscraper/__pycache__ directory

mugicha101 committed:
basic query functionality implemented for the texas website

mugicha101 committed:
minor edgecase handling

mugicha101 committed:
added basic csv output

mugicha101 committed:
added better address matching

mugicha101 committed:
handle a list of queries concurrently using threads to speed up responses

atoneyd committed:
Add routes for uploading/modifying and dowloading csv file

mugicha101 committed:
separated test code from webscraper.py into test.py

atoneyd committed:
Rewrite code to work for excel files

rebeccarozansky commented:
Getting some errors when running your code. I had to change the import statement: from backend.webscraper.webscraper t-> from webscraper. After that change, I am getting this:

```
Traceback (most recent call last):
  File "C:\Users\becky\Documents\cssg\hkf\backend\webscraper\test.py", line 3, in <module>
    from webscraper import handle_queries, Query
  File "C:\Users\becky\Documents\cssg\hkf\backend\webscraper\webscraper.py", line 7, in <module>
    from state_websites import state_websites, Response, Query
  File "C:\Users\becky\Documents\cssg\hkf\backend\webscraper\state_websites.py", line 36, in <module>
    class StateWebsite(object):
  File "C:\Users\becky\Documents\cssg\hkf\backend\webscraper\state_websites.py", line 43, in StateWebsite
    def get_page(self) -> str | None:
TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
```

rebeccarozansky committed:
Merge pull request #5 from cssgunc/api-send-file

Modify and Send excel file

pawagh commented:
Looks good to me

mugicha101 committed:
fix imports

mugicha101 committed:
Merge branch 'webscraper' of https://github.com/cssgunc/hkf into webscraper

mugicha101 committed:
refix imports

mugicha101 committed:
fix type error

mugicha101 committed:
fix SSL certification error by disabling verification

mugicha101 committed:
add comment pointing out SSL certification vulnerability

mugicha101 committed:
add comments

mugicha101 committed:
add info to run test

mugicha101 commented:
> Getting some errors when running your code. I had to change the import statement: from backend.webscraper.webscraper t-> from webscraper. After that change, I am getting this:
> 
> ```
> Traceback (most recent call last):
>   File "C:\Users\becky\Documents\cssg\hkf\backend\webscraper\test.py", line 3, in <module>
>     from webscraper import handle_queries, Query
>   File "C:\Users\becky\Documents\cssg\hkf\backend\webscraper\webscraper.py", line 7, in <module>
>     from state_websites import state_websites, Response, Query
>   File "C:\Users\becky\Documents\cssg\hkf\backend\webscraper\state_websites.py", line 36, in <module>
>     class StateWebsite(object):
>   File "C:\Users\becky\Documents\cssg\hkf\backend\webscraper\state_websites.py", line 43, in StateWebsite
>     def get_page(self) -> str | None:
> TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
> ```

This issue should be fixed, I replaced str | None with Optional[str] and also fixed the imports
